wdl-lint/src/rules/disallowed_declaration_name.rs

```diff
@@ -0,0 +1,190 @@
+fn check_decl_name(
+    state: &mut Diagnostics,
+    decl: &Decl,
+    exceptable_nodes: &Option<&'static [SyntaxKind]>,
+) {
+    let mut type_names = HashSet::new();
+    match decl.ty() {
+        Type::Ref(_) => return, // Skip type reference types (user-defined structs)
+        Type::Primitive(primitive_type) => {
+            match primitive_type.kind() {
+                // Skip File and String types as they cause too many false positives
+                PrimitiveTypeKind::File | PrimitiveTypeKind::String => return,
+                PrimitiveTypeKind::Boolean => {
+                    type_names.insert(primitive_type.to_string());
+                    type_names.insert("Bool".to_string());
+                }
+                PrimitiveTypeKind::Integer => {
+                    // Integer is shortened to Int in WDL
+                    type_names.insert(primitive_type.to_string());
+                    type_names.insert("Integer".to_string());
+                }
+                PrimitiveTypeKind::Float => {
+                    type_names.insert(primitive_type.to_string());
+                }
+                PrimitiveTypeKind::Directory => {
+                    type_names.insert(primitive_type.to_string());
+                    type_names.insert("Dir".to_string());
+                }
+            }
+        }
+        Type::Array(_) => {
+            type_names.insert("Array".to_string());
+        }
+        Type::Map(_) => {
+            type_names.insert("Map".to_string());
+        }
+        Type::Pair(_) => {
+            type_names.insert("Pair".to_string());
+        }
+        Type::Object(_) => {
+            type_names.insert("Object".to_string());
+        }
+    }
+
+    let ident = decl.name();
+    let name = ident.text();
+    for type_name in &type_names {
+        let type_lower = type_name.to_lowercase();
+
+        // Special handling for short type names (3 characters or less).
+        // These require word-based checks to avoid false positives.
+        if type_lower.len() <= 3 {
+            let words = split_to_words(name);
+
+            if words.contains(&type_lower) {
+                let diagnostic = decl_identifier_with_type(ident.span(), name, type_name);
+                state.exceptable_add(
+                    diagnostic,
+                    rowan::NodeOrToken::Node(decl.inner().to_owned()),
+                    exceptable_nodes,
+                );
+                return;
+            }
+        } else if name.to_lowercase().contains(&type_lower) {
+            let diagnostic = decl_identifier_with_type(ident.span(), name, type_name);
+            state.exceptable_add(
+                diagnostic,
+                rowan::NodeOrToken::Node(decl.inner().to_owned()),
+                exceptable_nodes,
+            );
+            return;
+        }
+    }
+}
+
+/// Split an identifier into words using convert_case
+fn split_to_words(identifier: &str) -> HashSet<String> {
```

Comment:
`convert_case::split` already returns a `Vec<&str>`, so I don't think we get anything by further collecting it into a `HashSet` since we're returning as soon as we find a match.

I would remove this function and just assign `words` above with `let words = convert_case::split(&identifier, &convert_case::Boundary::defaults())`.